packages/color-area/src/ColorArea.ts

```diff
@@ -0,0 +1,389 @@
+/*
+Copyright 2020 Adobe. All rights reserved.
+This file is licensed to you under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+
+import {
+    html,
+    SpectrumElement,
+    CSSResultArray,
+    TemplateResult,
+    property,
+    PropertyValues,
+    query,
+    streamingListener,
+} from '@spectrum-web-components/base';
+import { ColorHandle, ColorValue } from '@spectrum-web-components/color-handle';
+import '@spectrum-web-components/color-handle/sp-color-handle.js';
+import { TinyColor } from '@ctrl/tinycolor';
+
+import styles from './color-area.css.js';
+
+/**
+ * @element sp-color-area
+ */
+export class ColorArea extends SpectrumElement {
+    public static get styles(): CSSResultArray {
+        return [styles];
+    }
+
+    @property({ type: Boolean, reflect: true })
+    public disabled = false;
+
+    @property({ type: Boolean, reflect: true })
+    public focused = false;
+
+    @query('.handle')
+    private handle!: ColorHandle;
+
+    @property({ type: Number })
+    public get hue(): number {
+        return this._hue;
+    }
+
+    public set hue(value: number) {
+        const hue = Math.min(360, Math.max(0, value));
+        if (hue === this.hue) {
+            return;
+        }
+        const oldValue = this.hue;
+        const { s, v } = this._color.toHsv();
+        this._color = new TinyColor({ h: hue, s, v });
+        this._hue = hue;
+        this.requestUpdate('hue', oldValue);
+    }
+
+    private _hue = 0;
+
+    @property({ type: String })
+    public get value(): ColorValue {
+        return this._value.toHex();
```

Comment:
I was worried that `this.color` wouldn't return a hex value, I think? But I just tested it out and it seems fine to me actually. I'll change it to `this.color`.

Reply:
The goal is to have it also return whatever you gave it, IMO. This is because if your app uses hex, you want hex back, but if you app uses hsl, you want that back.